src/tools/search/searXNGSearch.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2025 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ToolEmitter, Tool, ToolInput, BaseToolRunOptions } from "@/tools/base.js";
+import { z } from "zod";
+import { Emitter } from "@/emitter/emitter.js";
+import { createURLParams } from "@/internals/fetcher.js";
+import { RunContext } from "@/context.js";
+import {
+  SearchToolOptions,
+  SearchToolOutput,
+  SearchToolResult,
+  SearchToolRunOptions,
+} from "@/tools/search/base.js";
+import { ValidationError } from "ajv";
+
+export interface SearxngToolOptions extends SearchToolOptions {
+  baseUrl?: string;
+  maxResults: number;
+}
+
+type SearxngToolRunOptions = SearchToolRunOptions;
+
+export interface SearxngToolResult extends SearchToolResult {}
+
+export class SearxngToolOutput extends SearchToolOutput<SearxngToolResult> {
+  constructor(public readonly results: SearxngToolResult[]) {
+    super(results);
+  }
+
+  static {
+    this.register();
+  }
+
+  createSnapshot() {
+    return {
+      results: this.results,
+    };
+  }
+
+  loadSnapshot(snapshot: ReturnType<typeof this.createSnapshot>) {
+    Object.assign(this, snapshot);
+  }
+}
+
+export class SearxngTool extends Tool<
+  SearxngToolOutput,
+  SearxngToolOptions,
+  SearxngToolRunOptions
+> {
+  name = "Web Search";
+  description = `Search for online trends, news, current events, real-time information, or research topics.`;
+
+  public readonly emitter: ToolEmitter<ToolInput<this>, SearxngToolOutput> = Emitter.root.child({
+    namespace: ["tool", "web", "search"],
```

Comment:
Not about the namespace...

`namespace: ["tool", "search", "google"]` is what we are using for Google Search.